apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/UpdateHistory.scala

```diff
@@ -245,7 +246,12 @@ class UpdateHistory(
 
           _ <- cleanUpDataAfterDomainMigration(newHistoryId)
 
-          _ <- deleteInvalidAcsSnapshots(newHistoryId)
+          _ <-
+            if (enableImportUpdateBackfill) {
+              deleteInvalidAcsSnapshots(newHistoryId)
```

Comment:
Change 1: do not attempt to delete corrupt snapshots if import update backfilling is disabled